### PR TITLE
Add prometheus metrics test into E2E

### LINKED ISSUE #5635: Add metrics testing for e2e
# 🚀 Feature Request

### Description

One thing an e2e test can look at is metrics data and use the data to determine pass/fail criteria. Here's a few interesting metrics to look at:

- `beacon_clock_time_slot`: should be the same as current slot
- `committee_cache_miss/committee_cache_hit`: should be <1%
- `hot_state_cache_miss/hot_state_cache_hit`: should be <1%
- `go_memstats_alloc_bytes`: should be lower than 1GB
- `p2p_message_failed_validation_total / p2p_message_received_total`
  - `beacon_aggregate_and_proof` should not be 100%
  - `beacon_block` should not be >20%
  - `committee_index_beacon_attestation` should not be >10%
- `bolt_bucket_keys`: should have the correct number of states and blocks saved in the db, this depends on how many epochs test runs

* The numbers are just an estimate, need more looking

### PR DESCRIPTION
**Pull request by 0xKiwi**
**What type of PR is this?**
Improvement

**What does this PR do? Why is it needed?**
This PR improves E2E testing by adding metric checks, to ensure breaking changes aren't added that cause an increased amount of ssz topic errors, and memory usage.

**Which issues(s) does this PR fix?**

Fixes #5635

### CODE REVIEW

File: endtoend/evaluators/metrics.go [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+	topic1             string
+	topic2             string
+	expectedComparison float64
+}
+
+var metricLessThanTests = []equalityTest{
+	{
+		name:  "memory usage",
+		topic: "go_memstats_alloc_bytes",
+		value: 100000000,
```
Comment (prestonvanloon) [inline endtoend/evaluators/metrics.go:46]:
```suggestion
		value: 100000000, // 100 Mb
```

File: endtoend/evaluators/metrics.go [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+}
+
+func getValueOfTopic(pageContent string, topic string) (int, error) {
+	// Adding a space to search exactly.
+	startIdx := strings.LastIndex(pageContent, topic+" ")
+	if startIdx == -1 {
+		return -1, fmt.Errorf("did not find requested text %s in %s", topic, pageContent)
+	}
+	endOfTopic := startIdx + len(topic)
+	fmt.Println(pageContent[startIdx:endOfTopic])
```
Comment (terencechain) [inline endtoend/evaluators/metrics.go:194]: is `fmt.Println` here intended or just for debug?

File: endtoend/evaluators/metrics.go [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+	topic1             string
+	topic2             string
+	expectedComparison float64
+}
+
+var metricLessThanTests = []equalityTest{
+	{
+		name:  "memory usage",
+		topic: "go_memstats_alloc_bytes",
+		value: 100000000, // 100 Mb
```
Comment (terencechain) [inline endtoend/evaluators/metrics.go:46]: 100MB only? I guess since it's only a few validators

File: endtoend/endtoend_test.go [hunk 4]
```diff
@@ -71,7 +71,8 @@ func runEndToEndTest(t *testing.T, config *types.E2EConfig) {
 	}
 	// Small offset so evaluators perform in the middle of an epoch.
 	epochSeconds := params.BeaconConfig().SecondsPerSlot * params.BeaconConfig().SlotsPerEpoch
-	genesisTime := time.Unix(genesis.GenesisTime.Seconds+int64(epochSeconds/2), 0)
+	// Adding a half slot here to ensure the requests are in the middle of an epoch.
+	genesisTime := time.Unix(genesis.GenesisTime.Seconds+int64(epochSeconds/2+(params.BeaconConfig().SecondsPerSlot/2)), 0)
```
Comment (nisdas) [inline endtoend/endtoend_test.go:75]: this is genesis time + half_epoch + half_slot . Can you clarify why you are doing this , this way ?

File: endtoend/evaluators/metrics.go (hunk 3 above)
Comment (nisdas) [reply endtoend/evaluators/metrics.go:46]: how do you get 100mb ? is there a better way to derive expected memory usage other than hardcoding it.

Comment (0xKiwi) [reply endtoend/evaluators/metrics.go:46]: I ran it and it seemed to be around 40mb normally. And yeah this is for only 64 validators so thats why its slow low.

File: endtoend/evaluators/metrics.go (hunk 2 above)
Comment (0xKiwi) [reply endtoend/evaluators/metrics.go:194]: My bad, thanks.

File: endtoend/endtoend_test.go (hunk 4 above)
Comment (0xKiwi) [reply endtoend/endtoend_test.go:75]: Sure thing

Comment (0xKiwi) [reply endtoend/endtoend_test.go]: Added clearer comments.

File: endtoend/long_minimal_e2e_test.go [hunk 5]
```diff
@@ -16,7 +16,7 @@ func TestEndToEnd_Long_MinimalConfig(t *testing.T) {
 	testutil.ResetCache()
 	params.UseMinimalConfig()
 
-	epochsToRun := 20
+	epochsToRun := 100
```
Comment (prestonvanloon) [inline endtoend/long_minimal_e2e_test.go:19]: This doesn't matter right? It gets set from ENV variable

```suggestion
	var epochsToRun int
```

Comment (0xKiwi) [reply endtoend/long_minimal_e2e_test.go:19]: Ah my bad was for testing

**Pull request merged**